OpenRA.Mods.Common/Widgets/SupportPowerTimerWidget.cs

```diff
@@ -21,23 +21,56 @@ public class SupportPowerTimerWidget : Widget
 	{
 		public readonly string Font = "Bold";
 		public readonly string Format = "{0}: {1}";
-		public readonly TimerOrder Order = TimerOrder.Descending;
+		public readonly string AlliesHeader = "Allies";
+		public readonly string EnemiesHeader = "Enemies";
+		public enum TimerOrder { EnemiesFirst = 0, AlliesFirst = 1 }
+		public readonly TimerOrder Order = TimerOrder.EnemiesFirst;
 
 		readonly IEnumerable<SupportPowerInstance> powers;
-		Pair<string, Color>[] texts;
+		readonly IEnumerable<SupportPowerInstance> powersAllies;
+		readonly IEnumerable<SupportPowerInstance> powersEnemies;
+		Pair<Player, Color>[] playersAllies;
+		Pair<string, Color>[] textsAllies;
+		Pair<Player, Color>[] playersEnemies;
+		Pair<string, Color>[] textsEnemies;
 
 		[ObjectCreator.UseCtor]
 		public SupportPowerTimerWidget(World world)
 		{
 			powers = world.ActorsWithTrait<SupportPowerManager>()
-				.Where(p => !p.Actor.IsDead && !p.Actor.Owner.NonCombatant)
+				.Where(p => !p.Actor.IsDead && !p.Actor.Owner.NonCombatant && p.Actor.Owner != world.LocalPlayer)
 				.SelectMany(s => s.Trait.Powers.Values)
 				.Where(p => p.Instances.Any() && p.Info.DisplayTimer && !p.Disabled);
+
+			powersAllies = powers.Where(p => world.LocalPlayer.IsAlliedWith(p.Instances[0].Self.Owner));
+			powersEnemies = powers.Where(p => !world.LocalPlayer.IsAlliedWith(p.Instances[0].Self.Owner));
 		}
 
 		public override void Tick()
 		{
-			texts = powers.Select(p =>
+			playersAllies = powersAllies.Select(p =>
```

Comment:
I suggest moving these queries into the constructor, so you don't have to rebuild them every tick before you iterate them.